16-practical-work-2/PRESENTATION.md

```diff
@@ -181,57 +185,296 @@ Bye.
+More details for this section in the [course material][course-material].
+
+### Group composition
+
+- 2 students per group
+- Create a GitHub Discussion to:
+  - Announce your group members
+  - Announce your idea (even a draft is fine)
+- **Do it as soon as possible before next week!**
+  - This helps us to plan the presentations
+
+![bg right:40%](https://images.unsplash.com/photo-1491911923017-19f90d8d7f83?fit=crop&h=720)
+
+## Idea validation
+
+<!-- _class: lead -->
+
+More details for this section in the [course material][course-material].
+
+### Idea validation
 
-[![bg right w:75%][practical-work-qr-code]][practical-work]
+- You must state your idea on your GitHub Discussion
+- We might ask you to change your idea if it is too simple or too complex
+- We will help you to find a good idea if needed
+- **Do it as soon as possible before next week!**
+
+![bg right:40%](https://images.unsplash.com/photo-1543075137-5a97903aaa7a?fit=crop&h=720)
+
+## Grading criteria
+
+<!-- _class: lead -->
+
+More details for this section in the [course material][course-material].
+
+### Grading criteria
+
+You can find all the grading criteria in the [course material][course-material]:
+
+- 0 point - The work is insufficient
+- 0.1 point - The work is done
+- 0.2 point - The work is well done (without the need of being perfect)
+
+Maximum grade: 25 points \* 0.2 + 1 = 6
+
+![bg right:40%](https://images.unsplash.com/photo-1645500010250-01b5bc42239e?fit=crop&h=720)
+
+## Constraints
+
+<!-- _class: lead -->
+
+More details for this section in the [course material][course-material].
+
+### Constraints
+
+- The application must be written in Java, compatible with Java 21
+- The application must be built using Maven with the `maven-shade-plugin` plugin
+- The application must use the picocli dependency
+- You can only use the Java classes seen in the course to process the files (you
+  can use other libraries to help you once the files are opened)
+- Your application must be slightly more complex and slightly different than the
+  examples presented during the course
+
+## Tips
+
+<!-- _class: lead -->
+
+More details for this section in the [course material][course-material].
+
+### Create diagrams
+
+- You can use any tools you want to create your diagrams:
+  - [PlantUML](https://plantuml.com/)
+  - [draw.io](https://draw.io/)
+  - etc.
+
+![bg right:40% contain](https://upload.wikimedia.org/wikipedia/commons/3/30/Plantuml_Logo.svg)
+
+### The POSIX standard
+
+> The Portable Operating System Interface (POSIX) standard is a family of
+> standards specified by the IEEE Computer Society for maintaining compatibility
+> between operating systems. POSIX defines both the system and user-level
+> application programming interfaces (APIs), along with command line shells and
+> utility interfaces, for software compatibility (portability) with variants of
+> Unix and other operating systems.
+>
+> <https://en.wikipedia.org/wiki/POSIX>
```

Comment:
I don't understand why there's an explanation of POSIX here

Reply:
The POSIX standard was, in fact, introduced in the "Define an application protocol" chapter last year along with the UNIX philosophy and the KISS principle as seen in this PR: https://github.com/heig-vd-dai-course/heig-vd-dai-course/pull/312/files#diff-678e1cc1973f0ceaaacaae0d277c9b7db8ea574704cc45a31604e434eb222e8cL273-L279. Since that PR, the UNIX philosophy and the KISS principle has been moved to the practical work 1 (here in the Tips section: <https://github.com/heig-vd-dai-course/heig-vd-dai-course/blob/main/07-practical-work-1/COURSE_MATERIAL.md#tips>.) to encourage students to do well and not over-complicated things.

As the POSIX standard was maybe a bit too soon to be introduced in the first part of the course, I wanted to give another "advice"/"reflection point" in this second practical work as done in the previous evaluation.

If I remember correctly, no other course speaks much about the POSIX and I think we can gain skills from that standard. However, I agree it comes here a bit from nowhere but I don't know where else to put it and even if it makes sense to speak about it at all.

Reply:
I agree with you that it comes a bit out of nowhere but I don't know where else to put it too. I think it's a good thing to leave this comment here, even for general culture.